codes/rust/chapter_sorting/counting_sort.rs

```diff
@@ -31,16 +31,16 @@ fn counting_sort_naive(nums: &mut [i32]) {
 // 完整实现，可排序对象，并且是稳定排序
 fn counting_sort(nums: &mut [i32]) {
     // 1. 统计数组最大元素 m
-    let m = *nums.into_iter().max().unwrap();
+    let m = *nums.iter().max().unwrap() as usize;
```

Comment:
Why `as usize`?

Reply:
as you can see in the context, m is itended to  be used as usize, in here we cast it to usize instead of multiple case in used place.

Reply:
as you can see in the context, m is itended to  be used as usize, in here we cast it to usize instead of multiple case in used place.